magento/pwa-studio
# PWA-2096: Track Checkout

## Description

1. "Shipping Information added" event is tracked  whenever a user fills in their shipping information on the checkout page
2. "Shipping method added" event  is tracked  whenever  whenever a user fills in their shipping method on the checkout page
3. "Billing information added" event is tracked whenever a user fills in their billing information on the checkout page
4. "Review order"  event is tracked  whenever a user clicks the "review order" button on the checkout page
5. "Place Order" and "Order confirmation page view" events are tracked  whenever a user completes their order and sees the order confirmation page
6. PlaceOrder Event includes the following attributes:
    * orderID
    * payments (paymentTransactionID, paymentAmount, paymentType, paymentCurrencyCode)
    * shipping (shippingMethod, shippingAmount)
    * promotionID
    * productArray
7. Out of scope: coupon code, gift card, gift options
8. Example extension subscribes to the events and has events available for processing


**Added checkout related events:**

1. Shipping Information added/updated
    * `CHECKOUT_SHIPPING_INFORMATION_ADDED`
    * `CHECKOUT_SHIPPING_INFORMATION_UPDATED`
2. Shipping method added/updated
    * `CHECKOUT_SHIPPING_METHOD_ADDED`
    * `CHECKOUT_SHIPPING_METHOD_UPDATED`
3. Billing information added/updated
    * `CHECKOUT_BILLING_INFORMATION_ADDED`
    * `CHECKOUT_BILLING_INFORMATION_UPDATED`
4. Review order button clicked event
    * `CHECKOUT_REVIEW_BUTTON_CLICKED`
5. Place order button clicked event
    * `CHECKOUT_PLACE_ORDER_BUTTON_CLICKED`
6. Order confirmation page view event was added in previous story - `ORDER_CONFIRMATION_PAGE_VIEW`


## Related Issue







Closes #[PWA-2096](https://jira.corp.magento.com/browse/PWA-2096).

## Acceptance





### Verification Stakeholders



### Specification




## Verification Steps






1. With items in cart, go to the checkout page
2. Fill out the shipping information and continue to next section
3. Verify that the sample extension logged the shipping information added event (with its contextual data)
4. Select the shipping method and continue to next section
5. Verify that the sample extension logged the shipping method added event (with its contextual data)
6. Fill out the billing method and address and continue to the next section
8. Verify that the sample extension logged the billing information added event (with its contextual data)
9. Click on Review order button
10. Edit shipping address, metho, and billing info and enter new information
11. For each edit, verify the sample extension logged the appropriate event with the correct data
12. Click on Place Order Button
13. Verify these steps for both Guest and Registered users.
14. Verify these steps for the zero checkout flow, i.e. using a gift card that exceeds total cost so you pay "zero" at checkout

#### Test scenario(s) for direct fix/feature





#### Test scenario(s) for any existing impacted features/areas





#### Test scenario(s) for any Magento Backend Supported Configurations





#### Is Browser/Device testing needed?





#### Any ad-hoc/edge case scenarios that need to be considered?




## Screenshots / Screen Captures (if appropriate)

## Breaking Changes (if any)





## Checklist



-   I have added tests to cover my changes, if necessary.
-   I have added translations for new strings, if necessary.
-   I have updated the documentation accordingly, if necessary.
